### ISSUE TITLE
Add support for milestones in release API

### ISSUE THREAD

**Issue by lazystone** [gitlab4j/gitlab4j-api#489]
I haven't found any way to provide list of milestones in the TagsApi.createRelease and updateRelease.

See https://gitlab.inoviagroup.se/help/api/releases/index.md#create-a-release for details

**Comment by gmessner** [comment]
Creating a release with milestones is part of the Releases API which is a recent addition to GitLab (introduced in 11.7).  Support for this will be added in the next release which should be completed in the next week or so.  We will be adding all methods found here: https://docs.gitlab.com/ce/api/releases/

**Comment by gmessner** [comment]
@lazystone 
GitLab4J-API release 4.14.8 is now available with support for the Release API (See ReleasesApi).  Please verify and close this issue when you have time.

**Comment by lazystone** [comment]
Hi @gmessner.

I tried new API - release update fails with 500. I couldn't find how you do json (de)serialization, but can it be that `tag_name` and `released_at` in ReleaseParams sent as `tagName` and `releaseAt`?

**Comment by lazystone** [comment]
Also `milestones` in the create/update call is an array of strings(The title of each milestone the release is associated with), but in ReleaseParams it's a Milestone object.

**Comment by gmessner** [comment]
@lazystone 
Concerning the update failing with a 500, it is not a JSON issue the (de)serialization is correct, there are integration tests that verify it is working correctly, and just to be sure I have added setting ```releasedAt``` to the test.  Here is the output of the logger to demonstrate that the JSON is correct, both in and out:
```
Dec 29, 2019 11:24:55 AM org.gitlab4j.api.utils.MaskingLoggingFilter log
INFO: 1 - Sending client request on thread main
1 > PUT http://192.168.1.6/api/v4/projects/1/releases/test%2Drelease%2F1%2E0%2E0
1 > Accept: application/json
1 > Content-Type: application/json
1 > PRIVATE-TOKEN: ********
{"tag_name":"test-release/1.0.0","description":"UPDATED: Test releases API release.","milestones":["Test Release 1.0.0 Milestone"],"released_at":"2019-12-29T19:24:55.204Z"}

Dec 29, 2019 11:24:55 AM org.gitlab4j.api.utils.MaskingLoggingFilter log
INFO: 1 - Received server response on thread main
1 < 200
1 < Cache-Control: max-age=0, private, must-revalidate
1 < Connection: keep-alive
1 < Content-Length: 2174
1 < Content-Type: application/json
1 < Date: Sun, 29 Dec 2019 19:24:54 GMT
1 < Etag: W/"e17b326f64485aaf504fcc9666671bfa"
1 < Referrer-Policy: strict-origin-when-cross-origin
1 < Server: nginx
1 < Strict-Transport-Security: max-age=31536000
1 < Vary: Origin
1 < X-Content-Type-Options: nosniff
1 < X-Frame-Options: SAMEORIGIN
1 < X-Request-Id: 1yGA8qgkVg2
1 < X-Runtime: 0.096607
{"name":"Test Release","tag_name":"test-release/1.0.0","description":"UPDATED: Test releases API release.","description_html":"\u003cp data-sourcepos=\"1:1-1:35\" dir=\"auto\"\u003eUPDATED: Test releases API release.\u003c/p\u003e","created_at":"2019-12-29T19:24:54.062Z","released_at":"2019-12-29T19:24:55.204Z","author":{"id":2,"name":"GitLab4J Tester","username":"gitlab4j","state":"active","avatar_url":"http://aa2a38aaf6c9/uploads/-/system/user/avatar/2/avatar.png","web_url":"http://aa2a38aaf6c9/gitlab4j"},"commit":{"id":"52f6bc4bdcaf972b3d2279eda7226bd62fa0a2e3","short_id":"52f6bc4b","created_at":"2019-12-28T22:04:56.000+00:00","parent_ids":["11528410bf31f488812fca6de0d6f738e6721821"],"title":"No longer needed.","message":"No longer needed.","author_name":"GitLab4J Tester","author_email":"gitlab4j@gitlab4j.org","authored_date":"2019-12-28T22:04:56.000+00:00","committer_name":"GitLab4J Tester","committer_email":"gitlab4j@gitlab4j.org","committed_date":"2019-12-28T22:04:56.000+00:00"},"upcoming_release":true,"milestones":[{"id":10,"iid":10,"project_id":1,"title":"Test Release 1.0.0 Milestone","description":null,"state":"active","created_at":"2019-12-29T19:24:54.338Z","updated_at":"2019-12-29T19:24:54.338Z","due_date":null,"start_date":null,"web_url":"http://aa2a38aaf6c9/gitlab4j/test-project/-/milestones/10"}],"commit_path":"/gitlab4j/test-project/commit/52f6bc4bdcaf972b3d2279eda7226bd62fa0a2e3","tag_path":"/gitlab4j/test-project/-/tags/test-release%2F1.0.0","assets":{"count":4,"sources":[{"format":"zip","url":"http://aa2a38aaf6c9/gitlab4j/test-project/-/archive/test-release/1.0.0/test-project-test-release-1.0.0.zip"},{"format":"tar.gz","url":"http://aa2a38aaf6c9/gitlab4j/test-project/-/archive/test-release/1.0.0/test-project-test-release-1.0.0.tar.gz"},{"format":"tar.bz2","url":"http://aa2a38aaf6c9/gitlab4j/test-project/-/archive/test-release/1.0.0/test-project-test-release-1.0.0.tar.bz2"},{"format":"tar","url":"http://aa2a38aaf6c9/gitlab4j/test-project/-/archive/test-release/1.0.0/test-project-test-release-1.0.0.tar"}],"links":[]},"_links":{"edit_url":"http://aa2a38aaf6c9/gitlab4j/test-project/-/releases/test-release%252F1.0.0/edit"}}
```

I have also fixed the milestones type (changed to String) in ```ReleaseParams``` and have added testing of this in the integration tests. This is available in GitLab4J-API release 4.14.9

If you still have problems with the new version please provide the entire stack trace for the exception and let me know the version of your GitLab server.

**Comment by lazystone** [comment]
Tested a couple of methods(get optional release, update release) of 4.14.9 against GitLab 12.6.0 - seems to work well now.

Thanks a lot and Happy New Year!
